cli/graph_util.rs

```diff
@@ -578,6 +619,53 @@ impl ModuleGraphBuilder {
       npm_resolver.resolve_pending().await?;
     }
 
+    let has_redirects_changed = graph.redirects.len() != initial_redirects_len;
+    let has_jsr_package_deps_changed =
+      graph.packages.package_deps_sum() != initial_package_deps_len;
+    let has_jsr_package_mappings_changed =
+      graph.packages.mappings().len() != initial_package_mappings_len;
+    let has_npm_packages_changed =
+      graph.npm_packages.len() != initial_npm_packages;
+
+    if has_redirects_changed
+      || has_jsr_package_deps_changed
+      || has_jsr_package_mappings_changed
+      || has_npm_packages_changed
```

Comment:
Todo: I'll add a test for each of these individual conditions before landing this PR.